KubeArmor/feeder/policyMatcher.go

```diff
@@ -12,440 +13,338 @@ import (
-							}
-						}
-					}
+			for _, src := range dir.FromSource {
+				if len(src.Path) > 0 {
+					fromSource = src.Path
+				} else if len(src.Directory) > 0 {
+					fromSource = src.Directory
+				} else {
+					continue
 				}
-			}
 
-			if len(secPolicy.Spec.File.MatchPatterns) > 0 {
-				//
+				match := newProcDirMatchPolicy(policyName, fromSource, dir)
+				matches.Policies = append(matches.Policies, match)
 			}
+		}
 
-			if len(secPolicy.Spec.Network.MatchProtocols) > 0 {
-				for _, proto := range secPolicy.Spec.Network.MatchProtocols {
-					if len(proto.FromSource) == 0 {
-						match := tp.MatchPolicy{}
-
-						match.PolicyName = secPolicy.Metadata["policyName"]
-
-						match.Severity = strconv.Itoa(proto.Severity)
-						match.Tags = proto.Tags
-						match.Message = proto.Message
-
-						match.Source = ""
-						match.Operation = "Network"
-
-						switch proto.Protocol {
-						case "TCP", "tcp":
-							match.Resource = "type=SOCK_STREAM"
-
-							matches.Policies = append(matches.Policies, match)
-						case "UDP", "udp":
-							match.Resource = "type=SOCK_DGRAM"
-
-							matches.Policies = append(matches.Policies, match)
-						case "ICMP", "icmp":
-							match.Resource = "type=SOCK_RAW protocol=1"
-
-						default:
-							continue
-						}
-
-						match.Action = proto.Action
-
-						matches.Policies = append(matches.Policies, match)
-					} else {
-						for _, src := range proto.FromSource {
-							if len(src.Path) > 0 {
-								match := tp.MatchPolicy{}
-
-								match.PolicyName = secPolicy.Metadata["policyName"]
-
-								match.Severity = strconv.Itoa(proto.Severity)
-								match.Tags = proto.Tags
-								match.Message = proto.Message
-
-								match.Source = src.Path
-								match.Operation = "Network"
-
-								switch proto.Protocol {
-								case "TCP", "tcp":
-									match.Resource = "type=SOCK_STREAM"
-
-								case "UDP", "udp":
-									match.Resource = "type=SOCK_DGRAM"
-
-								case "ICMP", "icmp":
-									match.Resource = "type=SOCK_RAW protocol=1"
-
-								default:
-									continue
-								}
-
-								match.Action = proto.Action
-
-								matches.Policies = append(matches.Policies, match)
-							} else if len(src.Directory) > 0 {
-								match := tp.MatchPolicy{}
-
-								match.PolicyName = secPolicy.Metadata["policyName"]
-
-								match.Severity = strconv.Itoa(proto.Severity)
-								match.Tags = proto.Tags
-								match.Message = proto.Message
-
-								match.Source = src.Directory
-								match.Operation = "Network"
-
-								switch proto.Protocol {
-								case "TCP", "tcp":
-									match.Resource = "type=SOCK_STREAM"
-
-								case "UDP", "udp":
-									match.Resource = "type=SOCK_DGRAM"
+		for _, patt := range secPolicy.Spec.Process.MatchPatterns {
+			if len(patt.Pattern) == 0 {
+				continue
+			}
 
-								case "ICMP", "icmp":
-									match.Resource = "type=SOCK_RAW protocol=1"
+			regexpComp, err := regexp.Compile(patt.Pattern)
```

Comment:
This is a proposal based on https://github.com/accuknox/KubeArmor/issues/143#issuecomment-862113552. It demanded a change in the types.go, adding Regexp field to `MatchPolicy` what I know isn't the right approach (draft for discussion).

As I've mentioned in the #143, Globbing patterns aren't Regular Expressions _per si_. AppArmor has its proper Globbing syntax and logic expansions which are very similar to bash ones with addition of some sugar. E.g.:

```
/dir/**  - match any file or directory in /dir/ or any directory below /dir/
/dir/**/ - match any directory in /dir/ or any directory below /dir/
```
Related to that we have two paths:
1. Using proper Regular Expressions `regexp.Regexp`
2. Using Globbing without syntax sugar `path.filepath.Glob` and/or `path.filepath.Match`

Following 2, we will have issues and lack of functionality. https://github.com/accuknox/KubeArmor/issues/143#issuecomment-861596803

For Globbing with syntax sugar, we can consider using [glob](https://github.com/gobwas/glob) which also promises better [performance](https://github.com/gobwas/glob#performance) than Regexp.